contract/models/contract_line.py

```diff
@@ -1299,7 +1297,7 @@ def fields_view_get(
     @api.multi
     def unlink(self):
         """stop unlink uncnacled lines"""
-        if not all(self.mapped('is_canceled')):
+        if not all(self.mapped(lambda r: r.is_canceled or r.display_type)):
```

Comment:
This is more performant with a `for` loop stopping at the first occurrence instead of forcing to process whole recordset.